src/platform-includes/sourcemaps/troubleshooting/javascript.mdx

```diff
@@ -1,167 +1,41 @@
 
-```
-sentry-cli releases files VERSION upload-sourcemaps script.min.js script.min.js.map --url-prefix '~/scripts'
-```
-
-You can also upload it with the fully qualified URL. For example:
-
-```
-sentry-cli releases files VERSION upload-sourcemaps . --url-prefix 'http://localhost:8000/scripts'
-```
-
-### Using the API
-
-You can alternately [use our API](/api/releases/update-an-organization-release-file/) to upload artifacts, following the same naming convention explained here.
-
-```shell
-curl -X POST \
-  https://sentry.io/api/0/organizations/ORG_SLUG/releases/VERSION/files/ \
-  -H 'Authorization: Bearer AUTH_TOKEN' \
-  -H 'content-type: multipart/form-data' \
-  -F file=@script.min.js.map \
-  -F 'name=~/scripts/script.min.js.map'
-```
-
-### Using the `~`
-
-The `~` is used in Sentry to replace the scheme and domain. It is not a glob!
-
-`http://example.com/dist/js/script.js` will match `~/dist/js/script.js` or `http://example.com/dist/js/script.js`
-
-but will NOT match `~/script.js`.
-
-## Multiple Origins
+Make sure all of the necessary files are available. For Sentry to de-minify your stack traces you must provide both the minified files (for example, app.min.js) and the corresponding source maps. In case the source map files do not contain your original source code (`sourcesContent`), you must additionally provide the original source files. Alternatively, sentry-cli will automatically embed the sources (if missing) into your source maps.
 
-It's not uncommon for a web application to be accessible at multiple origins. For example:
+## Verify you are building sourcemaps
 
-- Website is operable over both `https` and `http`
-- Geolocated web addresses: such as `https://us.example.com`, `https://eu.example.com`
-- Multiple static CDNs: such as `https://static1.example.com`, `https://static2.example.com`
-- Customer-specific domains/subdomains
+Bundlers and tools (like `tsc`) that generate code generally require you to manually set options in order for source maps to be generated.
 
-In this situation, **identical** JavaScript and source map files may be located at two or more distinct origins. In this situation we recommend using our special tilde (`~`) prefix on paths.
+If you followed our tool-specific guides, verify that you configured your tool to emit source maps and that the source maps contain your original source code inside the `sourcesContent` field.
 
-So for example, if you have the following:
+## Verify your source files contain debug ID injection snippets
 
-- https://static1.example.com/js/app.js
-- https://static2.example.com/js/app.js
+Inside the JavaScript files you uploaded to Sentry, search for code that roughly looks like `e._sentryDebugIds=e._sentryDebugIds||{}`. Having this code inside a bundle means that it will be able to be matched to a source file. Every bundle you deploy in your app needs to have this snippet in order to be correctly source mapped.
```

Comment:
Any chance that `e` could be minified/mangled to something else in user code? Would it be enough to just let them search for `_sentryDebugIds`?

Reply:
Had a similar thought while writing this. The reason we can't let them just search for `_sentryDebugIds` is that it's also inside SDK code. Since injection usually happens (or rather _should happen_) very late in the build process, it's likely not minified/mangled further anyhow so I would leave this as is. Wdyt?